includes/staticrequesthandler.php

```diff
@@ -18,19 +17,25 @@
     $url = $https ? 'https://' : 'http://';
     $url .= $host . $uri;
 
-
     if (is_callable($urlMapping)) {
         $path = $urlMapping($url);
     } else {
         $path = URLtoPath($url);
     }
 
-    if (!$path) {
+    if (! $path) {
         return false;
     }
 
     $cachePath = $cacheDir . DIRECTORY_SEPARATOR . $path;
-
+    $hasCachedFile = true;
+    if (file_exists($cachePath . '.html.gz')) {
```

Comment:
I think this needs to be a bit more robust. We shouldn't look for the gzipped version if there is no valid `Accept-Encoding` header. We should also always check the html file exists and calculate the ETag from that.